hack/ensure-azcli.sh

```diff
@@ -35,6 +35,7 @@ if [[ -z "$(command -v az)" ]]; then
     # Use --auth-mode "login" in az storage commands to use RBAC permissions of login identity. This is a well known ENV variable the Azure cli
     export AZURE_STORAGE_AUTH_MODE="login"
   else
-    az login --service-principal -u "${AZURE_CLIENT_ID}" -p "${AZURE_CLIENT_SECRET}" --tenant "${AZURE_TENANT_ID}" > /dev/null
+    echo "AZURE_FEDERATED_TOKEN_FILE environment variable must be set to path location of token file"
```

Comment:
do we want to provide another option for folks to run this script that are not running in a cluster with federated tokens?  maybe either `azure login --identity` or just `az login` when running manually?

Reply:
This could be done in a follow up as otherwise right now we are blocking PRs until this goes in

Reply:
Yeah, let's figure this out next week as a lower priority item, as you point out local dev is broken on main until we do so.

cc @nojnhuh @willie-yao @mboersma @Jont828